include/envoy/api/os_sys_calls_linux.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
```

Comment:
Can you add something like:
```
#ifndef __LINUX__
#error "Linux platform included on non-Linux"
#endif 
```

(or something like that). Same for the impl.

Reply:
Ok, done.